RELEASE.md

```diff
@@ -11,7 +11,7 @@ The `release` target will not build the Docker images - they should be built and
 The release process should normally look like this:
 1. Create a release branch
 2. Export the desired version into the environment variable `RELEASE_VERSION`
-3. Run `make clean release`
+3. Run `make release`
```

Comment:
I wonder if we should keep the clean and just use separate command for it. As in:
2. Run `make clean`
3. Export the desired ...
4. Run `make release`